From 70dd912b5bfe38f82095f8e973fbf42269e3e868 Mon Sep 17 00:00:00 2001
From: Bilux <i.bilux@gmail.com>
Date: Tue, 11 Feb 2020 20:19:39 +0100
Subject: [PATCH]  mtk: O: reintroduce atomic symbols

Reintroduce atomic symbols in libcutils for old MTK audio hals & Mali blobs
---
 libcutils/Android.bp |  1 +
 libcutils/atomic.c   | 19 +++++++++++++++++++
 2 files changed, 20 insertions(+)
 create mode 100644 libcutils/atomic.c

diff --git a/libcutils/Android.bp b/libcutils/Android.bp
index d00ff5f963e..74dc20530dd 100644
--- a/libcutils/Android.bp
+++ b/libcutils/Android.bp
@@ -56,6 +56,7 @@ cc_library {
     },
     host_supported: true,
     srcs: [
+        "atomic.c",
         "config_utils.c",
         "fs_config.cpp",
         "canned_fs_config.c",
diff --git a/libcutils/atomic.c b/libcutils/atomic.c
new file mode 100644
index 00000000000..990526ab5c7
--- /dev/null
+++ b/libcutils/atomic.c
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2017 The LineageOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define ANDROID_ATOMIC_INLINE
+
+#include <cutils/atomic.h>
